### PR TITLE
[Resolve Errors] [Review Warnings] Tables - full borders will only show when overflowing (horizontal scrolling active)

### LINKED ISSUE #547: [Resolve Errors] [Review Warnings] Tables - selectively enable borders based on if the table is overflowing
- `useIsOverflowing` hook already added to the app
- Requires ref forwarding on the Table container

### PR DESCRIPTION
**Pull request by shindigira**
closes #547 

Note: Requires https://github.com/cfpb/design-system-react/pull/358 to be merged in first

## Changes
- enhancement-style(tables): full borders will only show for overflowing
- chore(tables): removed the need for `showTableBorders`

## How to test this PR

1. `yarn install` to get the latest DSR update with the table ref change
2. use [all-errors-warnings-no-syntax-errors.csv](https://github.com/cfpb/sbl-frontend/files/15444623/all-errors-warnings-no-syntax-errors.csv) or any register with multifield errors that causes overflowing/horizontal scrolling
3. Verify that tables without overflowing don't have full borders, tables that do have overflowing/horizontal scrolling have full borders

## Screenshot
Screenshot 2024-05-25 at 11 47 57 AM

### CODE REVIEW

File: src/pages/Filing/FilingApp/FieldEntry.tsx [hunk 1]
```diff
@@ -104,6 +101,10 @@ function FieldEntry({
     setCurrentPage(inputNumber);
   };
 
+  // Selectively enable full table borders based on if the table's wrapper div is overflowing
+  const [tableDivReference, tableDivReferenceIsOverflowing] =
+    useIsOverflowing();
```
Comment (billhimmelsbach) [inline src/pages/Filing/FilingApp/FieldEntry.tsx:106]: Does this run on every table, even ones we know won't be wider than the screen? Like for single field?

**Review by billhimmelsbach — COMMENTED** [review]
Looking good! One quick perf question.

**Review by billhimmelsbach — APPROVED** [review]
Sounds good! Looking forward to the follow-up PR!

**Comment by natalia-fitzgerald** [comment]
@shindigira 
In the screenshot the bottom border is missing. It should be there if the other borders are present.

**Pull request merged**